core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -226,7 +225,15 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
             if (terms.getBucketsInternal().size() < getShardSize() || InternalOrder.isTermOrder(order)) {
                 thisAggDocCountError = 0;
             } else if (InternalOrder.isCountDesc(this.order)) {
-                thisAggDocCountError = terms.getBucketsInternal().get(terms.getBucketsInternal().size() - 1).docCount;
+                if (terms.getDocCountError() > 0) {
+                    // If there is an existing docCountError for this agg then
+                    // use this as the error for this aggregation
+                    thisAggDocCountError = terms.getDocCountError();
```

Comment:
maybe we should better name this variable (or add comments) to reflect that it is about the error that this particular agg might introduce for the global results? Right now it looks like it is about the error for this particular agg, so it feels wrong to not use `terms.getDocCountError()` all the time.